codegen/src/ClassCoordinates.scala

```diff
@@ -29,12 +52,29 @@ case class ClassCoordinates(
   }
 
   def filePath(implicit providerConfig: Config.ProviderConfig): FilePath = {
-    val moduleName = modulePackageParts.head
-    val moduleParts =
-      if (providerConfig.noncompiledModules.contains(moduleName))
-        s".${moduleName}" +: modulePackageParts.tail // A leading dot excludes a directory from scala-cli sources
-      else
-        modulePackageParts
-    FilePath(Seq("src") ++ moduleParts ++ Seq(s"${className}.scala"))
+    val moduleParts = modulePackageParts match {
+      case "index" :: tail => tail
```

Comment:
Other SDKs does not use "index" in code and I'd be in favor of following upstream conventions, unless we have a very good reason.

I personally find `index` package very confusing, and not something that helps with reusing Pulumi knowledge and examples from other languages.

Reply:
Maybe there is an opportunity here to incorporate provider functions and methods to this discussion, and we might want to create some `syntax` file and use `export`, WDYT?

Reply:
There wasn't and won't be any `index` in **package** - only in the directories structure - this is transparent from the point of view of users.
E.g. for provider `foo` with modules `index` and `bar` we would have:

1) foo/src/index/Resource1.scala:
```scala
package besom.api.foo

class Resource1
```

2) foo/src/bar/Resource2.scala:
```scala
package besom.api.foo.bar

class Resource2
```
Users would refer to the resources as `besom.api.foo.Resource1` and `besom.api.foo.bar.Resource2`.
No exports are necessary here

Reply:
I personally find it confusing and unexpected to have package structure be not reflected in directory structure, but it is not a strong opinion.

I'm certainly ok with a tradeoff, because of "difficult to introduce separate compilation of modules" you've pointed out.

It also simplifies the code, and I like that as well, I will adjust in a followup PR accordingly.

Reply:
the final PR uses "index" as suggested in the review #264